### ISSUE TITLE
Support custom attribute in element

### ISSUE THREAD

**Issue by riceyeh** [excalidraw/excalidraw#6420]
I am trying to integrate excalidraw in my AP. I have a problem. When an event, 'click' for example, occurs on an excalidraw element, I have no way to tell what object in my AP is associated with since I cannot decide the id of excalidraw's element. Hence, is it possible to support custom attributes in excalidraw just like the user-* attributes in HTML element.

**Comment by ad1992** [comment]
@riceyeh we already support `customData` for elemeents, you can read more about it [here](https://docs.excalidraw.com/docs/@excalidraw/excalidraw/api/props/#storing-custom-data-on-excalidraw-elements)

**Comment by riceyeh** [comment]
Is it possible to edit customData like we do for the stroke color when elements are selected?

**Comment by dwelle** [comment]
Yes, though we don't have good API for listening on specific changes like elements being selected. The way to accomplish that currently would be to subscribe to `onChange`, diff the passed `appState.selectedElementIds`, and update the element using `excalidrawAPI.updateScene(elements)`.
